sunpy/net/attr.py

```diff
@@ -112,28 +109,27 @@ def __getattr__(self, item):
         registry = self._attr_registry[self]
         # All the attribute names under that type(Attr)
         names = registry.name
-        if item in names:
+        try:
             # We return Attr(name_long) to create the Attr requested.
             return self(registry.name_long[names.index(item)])
-        else:
+        except ValueError:
             raise AttributeError(f'This attribute, {item} is not defined, please register it.')
 
     def __dir__(self):
         """
         To tab complete in Python we need to add to the `__dir__()` return.
         So we add all the registered values for this subclass of Attr to the output.
         """
-        custom_attrs = list(set(self._attr_registry[self].name))
+        custom_attrs = set(self._attr_registry[self].name)
         # "all" can be registered as a documentation helper, but isn't a valid attr
-        if "all" in custom_attrs:
-            custom_attrs.remove("all")
-        return super().__dir__() + custom_attrs
+        custom_attrs.discard("all")
```

Comment:
Does the discard ignore if "all" is not found or is "all" in there all the time?

Reply:
Yeah, of course. I removed the need for an explicit check, you can take a look in the [docs](https://docs.python.org/3.7/library/stdtypes.html#frozenset.discard).